week3/ex4_permutations.py

```diff
@@ -0,0 +1,31 @@
+
+
+def all_perms_generator(elements):
+    if len(elements) <= 1:
+        yield elements
+    else:
+
+        for perm in all_perms_generator(elements[1:]):
+
+            for i in range(len(elements)):
+                ret = tuple(perm[:i]) + tuple(elements[0:1]) + tuple(perm[i:])
+                yield ret
+
+
+def permutations(elements):
```

Comment:
This function returns a list, but it should be a generator/iterator.
Please make it generator/iterator.